src/component.js

```diff
@@ -43,12 +43,150 @@ const required = (name) => {
+ * @property {() => void} clear - Function to clear the announcer.
+ * @property {() => void} disable - Function to disable the announcer.
+ * @property {() => void} enable - Function to enable the announcer.
+ * @property {(delay: number) => void} pause - Function to pause the announcer.
+ * @property {(message: string) => void} polite - Function to politely announce a message.
+ * @property {(message: string, politeness: 'off'|'polite'|'assertive') => void} announce - Function to announce a message with specified politeness.
+ * @property {() => void} stop - Function to stop the announcer.
+ * @property {(v: boolean) => void} toggle - Function to toggle the announcer.
+ *
+ * @typedef {Object} BlitsElementConfig
+ * @property {BlitsElement} parent - The parent element of this element.
+ * @property {any} node - The node object for the element.
+ *
+ * @typedef {Object} BlitsElementProps
+ * @property {boolean} __textnode - Indicates if the element is a text node.
+ * @property {boolean} __layout - Indicates if the element is a layout node.
+ * @property {BlitsElement} element - The element to which the props belong?? Do we need this?
+ * @property {BlitsElementConfig} config - Configuration object for the element.
+ * @property {Object<string, any>} props - The props object containing the properties of the element.
+ * @property {Object<string, any>} raw - The raw input props.
+ * @property {Object<string, any>} scheduledTransitions - Tracks transitions by property name.
+ *
+ * @typedef {Object} BlitsElement
+ * @property {BlitsComponent} component - Reference to the owning Blits component.
+ * @property {BlitsElementConfig} config - Configuration object for the element.
+ * @property {number} counter - Unique counter used for shader workarounds. FIXME?
+ * @property {string[]} effectNames - Names of active shader effects.
+ * @property {any} node - The underlying renderer node (e.g., WebGL node or text node).
+ * @property {BlitsElementProps} props - Proxy-like object containing transformed props.
+ * @property {any[]} children - WVB I cant see this populated? Filtered list of children owned by this element. FIXME?
+ * @property {any} parent - WVB Shortcut to the parent CoreNode?
+ * @property {number} nodeId - ID of the CoreNode, if available.
+ * @property {string|null} ref - Ref name (if defined).
+ * @property {function(Object):void} populate - Initializes the element with props and hooks.
+ * @property {function(string, any):void} set - Updates a single property.
+ * @property {function(string, any, Object):void} animate - Animates a property with transition options.
+ * @property {function():void} destroy - Destroys the underlying node and cancels transitions.
+ * @property {function(any): any} triggerLayout - Triggers a layout update for the element.
+ *
+ * @typedef {Object} BlitsLifecycle
+ * @property {BlitsComponent} component - The component WVB do we need this here again?
```

Comment:
yes it's the scope and this makes sure lifecycle scope is unique per component instance

Reply:
OK - will remove comment 😄